estate/tests/test_estate_property.py

```diff
@@ -0,0 +1,57 @@
+    def test_001_sell_property_whithout_offers(self):
+        estate_property_form = self.create_estate_property()
+        estate_property_form.status = "new"
+        estate_property = estate_property_form.save()
+        with self.assertRaisesRegex(UserError, "You cannot sell a property without offers."):
+            estate_property.action_sold_property()
+
+    def test_002_offer_creation_to_sold_property(self):
+        estate_property_form = self.create_estate_property()
+        estate_property_form.status = "sold"
+        estete_property = estate_property_form.save()
+        with self.assertRaisesRegex(UserError, "You cannot add an offer to a property that has already been sold."):
+            with Form(estete_property) as estate_property_form:
+                with estate_property_form.offer_ids.new() as offer:
+                    offer.partner_id = self.env.ref("base.res_partner_12")
+                    offer.price = 1500
+                    offer.validity = 3
+                    offer.date_deadline = fields.Date.add(fields.Date.today(), days=offer.validity)
+
+    def test_003_sell_property_whit_offer_received(self):
+        estate_property_form = self.create_estate_property()
+        estate_property_form.status = "offer_received"
+        estate_property = estate_property_form.save()
+        with self.assertRaisesRegex(UserError, "You cannot sell a property without accepting an offer."):
+            estate_property.action_sold_property()
+        self.assertEqual(estate_property.status, "sold", "The property status wasn't update.")
+        with Form(estate_property) as estate_property_form:
+            with estate_property_form.offer_ids.new() as offer:
+                offer.partner_id = self.env.ref("base.res_partner_12")
+                offer.price = 1500
+                offer.validity = 3
+                offer.date_deadline = fields.Date.add(fields.Date.today(), days=offer.validity)
+        estate_property = estate_property_form.save()
+        estate_property.offer_ids.action_accept_offer()
+        estate_property.action_sold_property()
+        self.assertEqual(estate_property.status, "sold", "The property status wasn't update.")
+
+    def test_004_uncheck_garden(self):
+        estate_property_form = self.create_estate_property()
+        estate_property_form.garden = True
+        estate_property = estate_property_form.save()
+        self.assertEqual(estate_property_form.garden_area, 10, "The garden area wasn't updated.")
+        self.assertEqual(estate_property_form.garden_orientation, "north", "The garden orientation wasn't updated.")
+        with Form(estate_property) as estate_property_form:
+            estate_property_form.garden = False
+        estate_property = estate_property_form.save()
+        self.assertEqual(estate_property.garden_area, 10, "The garden area wasn't updated.")
+        self.assertEqual(estate_property.garden_orientation, "north", "The garden orientation wasn't updated.")
```

Comment:
The message at the end of `assert` method is shown when the assert fails. In this case, you're checking that the values are still the same, so the message should be: "The value has been updated."

```suggestion
        self.assertEqual(estate_property.garden_area, 10, "The garden area has been updated.")
        self.assertEqual(estate_property.garden_orientation, "north", "The garden orientation has been updated.")
```